packages/eas-cli/src/commands/build/resign.ts

```diff
@@ -0,0 +1,292 @@
+import { Platform } from '@expo/eas-build-job';
+import { CredentialsSource, EasJson, EasJsonAccessor, EasJsonUtils } from '@expo/eas-json';
+import { Flags } from '@oclif/core';
+import assert from 'assert';
+import chalk from 'chalk';
+
+import { handleDeprecatedEasJsonAsync } from '.';
+import { waitForBuildEndAsync } from '../../build/build';
+import { ensureIosCredentialsForBuildResignAsync } from '../../build/ios/credentials';
+import { prepareCredentialsToResign } from '../../build/ios/prepareJob';
+import { listAndSelectBuildOnAppAsync } from '../../build/queries';
+import { printBuildResults, printLogsUrls } from '../../build/utils/printBuildInfo';
+import EasCommand from '../../commandUtils/EasCommand';
+import { ExpoGraphqlClient } from '../../commandUtils/context/contextUtils/createGraphqlClient';
+import { EasNonInteractiveAndJsonFlags } from '../../commandUtils/flags';
+import { EasPaginatedQueryFlags } from '../../commandUtils/pagination';
+import { CredentialsContext } from '../../credentials/context';
+import {
+  BuildFragment,
+  BuildMode,
+  BuildStatus,
+  BuildWorkflow,
+  DistributionType,
+  IosJobOverridesInput,
+  ProjectArchiveSourceType,
+  StatuspageServiceName,
+} from '../../graphql/generated';
+import { BuildMutation } from '../../graphql/mutations/BuildMutation';
+import { BuildQuery } from '../../graphql/queries/BuildQuery';
+import { toAppPlatform } from '../../graphql/types/AppPlatform';
+import Log from '../../log';
+import { requestedPlatformDisplayNames, selectPlatformAsync } from '../../platform';
+import { resolveXcodeBuildContextAsync } from '../../project/ios/scheme';
+import { resolveTargetsAsync } from '../../project/ios/target';
+import { getOwnerAccountForProjectIdAsync } from '../../project/projectUtils';
+import { printJsonOnlyOutput } from '../../utils/json';
+import { maybeWarnAboutEasOutagesAsync } from '../../utils/statuspageService';
+
+interface BuildResignFlags {
+  json: boolean;
+  nonInteractive: boolean;
+  offset?: number;
+  limit?: number;
+  platform?: Platform;
+  profile?: string;
+  maybeBuildId?: string;
+  wait: boolean;
+}
+interface RawBuildResignFlags {
+  json: boolean;
+  'non-interactive': boolean;
+  offset: number | undefined;
+  limit: number | undefined;
+  platform: 'android' | 'ios' | undefined;
+  profile: string | undefined;
+  wait: boolean;
+  'build-id': string | undefined;
+}
+
+export default class BuildResign extends EasCommand {
+  static override description = 'resign a build archive';
+
+  static override flags = {
+    platform: Flags.enum({
+      char: 'p',
+      options: ['android', 'ios'],
```

Comment:
Maybe we should drop `android` for now?
```suggestion
      options: ['ios'],
```

Reply:
I added that to avoid introducing breaking change when we add the android option, let's say someone has `eas build:resign --non-interactive` in their CI, but after we add the android param it would have to be required, so it would break those workflows.